src/Application/Service/Weaver.php

```
@@ -259,7 +259,16 @@ private function extractionRequest(array $turns): LlmRequest
                 $this->graphStore()->nodesByKind(NodeKind::Project, 12),
             ),
-            known: array_map(static fn ($n): string => $n->title, $this->graphStore()->graph(30, NodeKind::personalKinds())['nodes']),
+            // Titles carry their KIND. Without it the model reused a title but
+            // refiled it — the upsert keys on (kind, title), so "Мурчик" the
+            // person and "Мурчик" the topic became two nodes, and both were then
+            // read back to the assistant as separate things. weave:dedup cannot
+            // help: it refuses cross-kind merges, and rightly so, since person
+            // "Anna" and project "Anna" are genuinely different.
+            known: array_map(
+                static fn ($n): string => $n->title . ' (' . $n->kind->value . ')',
+                $this->graphStore()->graph(30, NodeKind::personalKinds())['nodes'],
```

PHPStan (2.2.8)

[error] Call to an undefined static method Semitexa\Weave\Domain\Enum\NodeKind::personalKinds().


(staticMethod.notFound)